src/trading/Topics.sol

```diff
@@ -0,0 +1,55 @@
+// Copyright (C) 2015 Forecast Foundation OU, full GPL notice in LICENSE
+
+pragma solidity ^0.4.13;
+
+import 'ROOT/libraries/Typed.sol';
+import 'ROOT/factories/IterableMapFactory.sol';
+import 'ROOT/libraries/token/VariableSupplyToken.sol';
+import 'ROOT/libraries/math/SafeMathUint256.sol';
+import 'ROOT/IController.sol';
+import 'ROOT/libraries/Initializable.sol';
+import 'ROOT/trading/ITopics.sol';
+
+
+contract Topics is DelegationTarget, Typed, Initializable, ITopics {
+    using SafeMathUint256 for uint256;
+
+    // Mapping of topic to popularity
+    IIterableMap private topics;
+
+    // FIXME: topics needs to be sorted on-chain, which means we need a sorted linked list collection with hinted sorting at insertion/update time instead of a simple iterableMap
+    function initialize(IController _controller) public beforeInitialized returns (bool) {
+        endInitialization();
+        controller = _controller;
```

Comment:
As this is a `DelegationTarget`, it already has a `controller` set, so this shouldn't be necessary.  I believe the next line (instantiating the map) is still necessary, so I think we need to keep `initialize` but it can be parameterless.